src/zarr/api/asynchronous.py

```diff
@@ -1120,8 +1119,10 @@ async def empty_like(
     retrieve data from an empty Zarr array, any values may be returned,
     and these are not guaranteed to be stable from one access to the next.
     """
-    like_kwargs = _like_args(a, kwargs)
-    return await empty(**like_kwargs)
+    like_kwargs = _like_args(a) | kwargs
+    if isinstance(a, (AsyncArray | Array)):
+        like_kwargs.setdefault("fill_value", a.metadata.fill_value)
+    return await empty(**like_kwargs)  # type: ignore[arg-type]
```

Comment:
Why the type ignore here? I guess the `LikeArgs` typed dict might solve that.

Reply:
Like many of the v2-era creation routines, `empty` has a nearly [untyped signature](https://github.com/zarr-developers/zarr-python/blob/f6742368229b8817a50335c28bb19a10e7a778e9/src/zarr/api/synchronous.py#L1129). The payoff for adding static typing to this particular invocation of `empty` is low until we add a fully typed signature to `empty` itself, which I'm not interested in doing in this PR